src/services/plugin.service.ts

```diff
@@ -12,6 +13,37 @@ export class PluginService {
     if (this.plugins[pluginName]) {
       throw new Error(`${pluginName} already exists as a plugin.`);
     }
-    return (this.plugins[pluginName] = new PluginLoader(pluginName, args) as IPlugin);
+
+    const reference = (this.plugins[pluginName] = new PluginLoader(pluginName, args) as IPlugin);
+
+    this.registerAliases(pluginName);
+
+    return reference;
+  }
+
+  registerAliases(pluginName: string, _args?: any): void {
+    const aliases = this.plugins[pluginName].pluginAlias;
+
+    if (aliases === undefined) {
+      this.aliases[pluginName] = pluginName;
+    } else {
+      if (!aliases.includes(pluginName)) {
+        if (this.aliases[pluginName] !== undefined) {
+          throw new Error(
+            `Duplicate alias detected: ${pluginName} is claiming its primary alias ${pluginName}, prevouisly claimed by ${this.aliases[pluginName]}.`
+          );
+        }
+        this.aliases[pluginName] = pluginName;
+      }
+
+      aliases.forEach((alias: string) => {
+        if (this.aliases[alias] !== undefined) {
+          throw new Error(
+            `Duplicate alias detected: ${pluginName} is trying to claim ${alias}, but ${this.aliases[alias]} has already claimed it.`
+          );
+        }
```

Comment:
duplicate code.

Reply:
While they both check for duplicates, they give different feedback.  I could put it in the function but it would literally only return a bool value.